core/rawdb/database.go

```diff
@@ -148,15 +151,74 @@ func NewDatabase(db ethdb.KeyValueStore) ethdb.Database {
 	}
 }
 
+func ReadOffSetOfCurrentAncientFreezer(db ethdb.KeyValueReader) uint64 {
+	offset, _ := db.Get(offSetOfCurrentAncientFreezer)
+	if offset == nil {
+		return 0
+	}
+	return new(big.Int).SetBytes(offset).Uint64()
+}
+
+func ReadOffSetOfLastAncientFreezer(db ethdb.KeyValueReader) uint64 {
+	offset, _ := db.Get(offSetOfLastAncientFreezer)
+	if offset == nil {
+		return 0
+	}
+	return new(big.Int).SetBytes(offset).Uint64()
+}
+
+func WriteOffSetOfCurrentAncientFreezer(db ethdb.KeyValueWriter, offset uint64) {
+	if err := db.Put(offSetOfCurrentAncientFreezer, new(big.Int).SetUint64(offset).Bytes()); err != nil {
+		log.Crit("Failed to store offSetOfAncientFreezer", "err", err)
+	}
+}
+func WriteOffSetOfLastAncientFreezer(db ethdb.KeyValueWriter, offset uint64) {
+	if err := db.Put(offSetOfLastAncientFreezer, new(big.Int).SetUint64(offset).Bytes()); err != nil {
+		log.Crit("Failed to store offSetOfAncientFreezer", "err", err)
+	}
+}
+
+//NewFreezerDb only create a freezer without statedb.
+func NewFreezerDb(db ethdb.KeyValueStore, frz, namespace string, readonly bool, lastOffSet, newOffSet uint64) (*freezer, error) {
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	// Create the idle freezer instance, this operation should be atomic to avoid mismatch between offset and acientDB.
+	frdb, errors := newFreezer(frz, namespace, readonly)
+	if errors != nil {
+		return nil, errors
+	}
+	frdb.offset = newOffSet
+	offsetBatch := db.NewBatch()
+	WriteOffSetOfCurrentAncientFreezer(offsetBatch, newOffSet)
```

Comment:
It is not proper to do `WriteOffSetOfCurrentAncientFreezer` inside `NewFreezerDb`.

Reply:
Separating this function is to ensure that two kinds of offsets are written into kvdb by the time creating ancient freezer. Any suggestion for this out of place you mentioned?

Reply:
I know why to separate the function. `WriteOffSetOfCurrentAncientFreezer ` should be placed in the main logic of block prune.

Reply:
Done